edgedb-protocol/src/serialization/decode/queryable/scalars.rs

```diff
@@ -0,0 +1,210 @@
+use crate::queryable::{Queryable, DescriptorContext, DescriptorMismatch};
+
+use crate::errors::DecodeError;
+use crate::codec;
+use crate::descriptors::TypePos;
+use crate::model::{Json, Uuid, Duration, LocalDate, LocalTime, LocalDatetime, BigInt, Decimal};
+use crate::serialization::decode::RawCodec;
+use std::time::SystemTime;
+
+fn check_scalar(ctx: &DescriptorContext, type_pos: TypePos, type_id: Uuid, name: &str) -> Result<(), DescriptorMismatch> {
+    use crate::descriptors::Descriptor::{Scalar, BaseScalar};
+    let desc = ctx.get(type_pos)?;
+    match desc {
+        Scalar(scalar) => {
+            return check_scalar(ctx, scalar.base_type_pos, type_id, name);
+        }
+        BaseScalar(base) if base.id == type_id => {
+            return Ok(());
+        }
+        _ => {}
+    }
+    Err(ctx.wrong_type(desc, name))
+}
+
+impl Queryable for String {
+    fn decode(buf: &[u8]) -> Result<Self, DecodeError> {
+        RawCodec::decode(buf)
+    }
+    fn check_descriptor(ctx: &DescriptorContext, type_pos: TypePos)
+        -> Result<(), DescriptorMismatch>
+    {
+        check_scalar(ctx, type_pos, codec::STD_STR, "str")
+    }
+}
+
+impl Queryable for Json {
+    fn decode(buf: &[u8]) -> Result<Self, DecodeError> {
+        RawCodec::decode(buf)
+    }
+    fn check_descriptor(ctx: &DescriptorContext, type_pos: TypePos)
+        -> Result<(), DescriptorMismatch>
+    {
+        check_scalar(ctx, type_pos, codec::STD_JSON, "json")
+    }
+}
+
+impl Queryable for Vec<u8> {
+    fn decode(buf: &[u8]) -> Result<Self, DecodeError> {
+        RawCodec::decode(buf)
+    }
+    fn check_descriptor(ctx: &DescriptorContext, type_pos: TypePos)
+        -> Result<(), DescriptorMismatch>
+    {
+        check_scalar(ctx, type_pos, codec::STD_BYTES, "bytes")

```

Comment:
Perhaps we should use the `std::` prefix here

```suggestion
        check_scalar(ctx, type_pos, codec::STD_BYTES, "std::bytes")

```